db/indextest/indextest_test.go

```diff
@@ -46,38 +46,40 @@ func TestRoleQuery(t *testing.T) {
 			defer database.Close(ctx)
 
 			n1QLStore, reset, err := setupN1QLStore(database.Bucket, testCase.isServerless)
-			assert.NoError(t, err, "Unable to get n1QLStore for testBucket")
+			require.NoError(t, err, "Unable to get n1QLStore for testBucket")
 			defer func(n1QLStore base.N1QLStore, isServerless bool) {
 				err := reset(n1QLStore, isServerless)
-				assert.NoError(t, err, "Reset fn shouldn't return error")
+				require.NoError(t, err, "Reset fn shouldn't return error")
 			}(n1QLStore, testCase.isServerless)
 
 			authenticator := database.Authenticator(ctx)
-			assert.NotNil(t, authenticator, "database.Authenticator(ctx) returned nil")
+			require.NotNil(t, authenticator, "database.Authenticator(ctx) returned nil")
 
 			// Add roles
 			for i := 1; i <= 5; i++ {
 				role, err := authenticator.NewRole(fmt.Sprintf("role%d", i), base.SetOf("ABC"))
-				assert.NoError(t, err, "Error creating new role")
-				assert.NoError(t, authenticator.Save(role))
+				require.NoError(t, err, "Error creating new role")
+				require.NoError(t, authenticator.Save(role))
 			}
 
 			// Delete 1 role
 			role1, err := authenticator.NewRole("role1", base.SetOf("ABC"))
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			err = authenticator.DeleteRole(role1, false, 0)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 
 			// Standard query
 			results, queryErr := database.QueryRoles(ctx, "", 0)
-			assert.NoError(t, queryErr, "Query error")
+			require.NoError(t, queryErr, "Query error")
+			defer func() {
+				require.NoError(t, results.Close())
+			}()
 			var row map[string]interface{}
 			rowCount := 0
 			for results.Next(&row) {
 				rowCount++
 			}
-			assert.Equal(t, 4, rowCount)
-			assert.NoError(t, results.Close())
+			require.Equal(t, 4, rowCount)
```

Comment:
This looks like it should still be an assert.

Reply:
It's now the last line, and if something was added to the test I wouldn't want it to continue. I don't have strong feelings though.